lib/wpc/wpc.c

```diff
@@ -880,6 +880,43 @@ app_res_e WPC_remote_scratchpad_update(app_addr_t destination_address, uint8_t s
     return convert_error_code(SCRATCHPAD_UPDATE_REMOTE_ERROR_CODE_LUT, res);
 }
 
+/* Error code LUT for target scratchpad write */
+static const app_res_e TARGET_SCRATCHPAD_ERROR_CODE_LUT[] = {
+    APP_RES_OK,               // 0
+    APP_RES_NODE_NOT_A_SINK,  // 1
+    APP_RES_INVALID_VALUE,    // 2
+    APP_RES_ACCESS_DENIED     // 3
+};
+
+app_res_e WPC_write_target_scratchpad(uint8_t target_sequence,
+                                      uint16_t target_crc,
+                                      uint8_t action,
+                                      uint8_t param)
+{
+    int res = msap_scratchpad_target_write_request(
```

Comment:
This is supposed to return negative value in case of error. But LUT below only handles position values (0...3)